pkg/cluster/update.go

```diff
@@ -104,6 +105,10 @@ func (u *simpleUpgrader) waitForNewNodes(ctx context.Context, cs *api.OpenShiftM
 				if err != nil {
 					return err
 				}
+				blob = append(blob, vmInfo{ComputerName: hostname, ScalesetHash: ssHashes[agent.Name]})
+				if err := u.updateBlob(blob); err != nil {
+					return err
+				}
```

Comment:
You are iterating all agent pool profiles and all vms here.  I don't believe it would be a difficult addition to check for vms that are in blob but have now gone away and remove them from blob.

In the long run, for testability I think we should separate the waitForReady, "record hash for vm we know we just created" and "delete old vms from blob" pieces into 3 sub-functions (probably all called from around here for efficiency); a follow-up for this would be good.